### PR TITLE
Feat button component

### PR DESCRIPTION
**Pull request by abdelazizsakhi**
- adding tslint properties

### CODE REVIEW

**Review by flowg — COMMENTED** [review]
Why is there a `yarn.lock` ? I thought we were going with NPM...
I actually have no problem with that but I'm afraid there might be some issues down the line... maybe they don't resolve dependencies the same way on some very obscure use-cases...
Am I being paranoid or ... ?

File: libs/ngx-dsfr/button/src/lib/testing/button.component.spec.ts [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+      expect(error).toBe(EMPTY_LABEL_ERROR);
+    }
+  });
+
+  describe(' all required properties are provided, ', () => {
+    beforeEach(() => {
+      testHost.testLabel = testLabel;
+    });
+
+    describe(' and inline is set to false, ', () => {
```
Comment (flowg) [inline libs/ngx-dsfr/button/src/lib/testing/button.component.spec.ts:78]: I don't see any reason to add a level of separation per use-case for this Component. So we should be able to make it with only 1 nested `describe`, the one for ' all required properties are provided, '.

Therefore, I suggest you merge the `beforeEach` of lines 73 and 79.

For the rest of the test cases, you can pretty much follow what I did for the `DsfrLinkComponent` and adapt. For instance, there should be a unit test testing what happens when a consumer wants to use icons with this Component, etc.

**Review by flowg — CHANGES_REQUESTED** [review]

File: .eslintrc.json [hunk 2]
```diff
@@ -12,14 +12,347 @@
             "enforceBuildableLibDependency": true,
             "allow": [],
             "depConstraints": [
+              {
```
[the remark this answers is not in the document]
Comment (abdelazizsakhi) [reply .eslintrc.json:15]: It's just some properties that i judge necessary but you can challenge it or we can modify it together.

File: libs/ngx-dsfr/button/src/lib/testing/button.component.spec.ts [hunk 3]
```diff
@@ -30,7 +28,7 @@ describe('DsfrButtonComponent', () => {
   let dsfrButtonHarness: DsfrButtonHarness;
 
   const testLabel = 'Test Label';
-  const testTitle = 'Test Title';
+  const testIcon = 'fr-icon-checkbox-circle-line'
```
Comment (flowg) [inline libs/ngx-dsfr/button/src/lib/testing/button.component.spec.ts:31]: `const testIcon = 'checkbox-circle-line';`

At this level, should be just the name of the icon, not the CSS class that the DSFR expects.

File: libs/ngx-dsfr/button/src/lib/testing/button.component.spec.ts [hunk 4]
```diff
@@ -7,20 +7,18 @@ import { HarnessLoader } from '@angular/cdk/testing';
 import { TestbedHarnessEnvironment } from '@angular/cdk/testing/testbed';
 
 /**
- * 3rd-party imports
+ * 3rd imports
```
Comment (flowg) [inline libs/ngx-dsfr/button/src/lib/testing/button.component.spec.ts:10]: `3rd-party imports` 😉

File: libs/ngx-dsfr/button/src/lib/testing/button.component.spec.ts [hunk 5]
```diff
@@ -61,40 +59,42 @@ describe('DsfrButtonComponent', () => {
...
-         * We're retrieving the Test Harness HERE since we can now,
-         * without any problem, trigger the Change Detection mechanism
-         *
-         * WARNING: Triggers the Change Detection mechanism
-         */
-        dsfrButtonHarness = await harnessLoader.getHarness<DsfrButtonHarness>(
-          DsfrButtonHarness
-        );
-      });
+    it('should set title identic to label when no title provided', async () => {
```
Comment (flowg) [inline libs/ngx-dsfr/button/src/lib/testing/button.component.spec.ts:80]: 'identical'

File: libs/ngx-dsfr/button/src/lib/testing/button.component.spec.ts [hunk 6]
```diff
@@ -61,40 +59,42 @@ describe('DsfrButtonComponent', () => {
...
+      const title: string | null = await dsfrButtonHarness.getButtonAttribute('title');
+      const label: string | null = await (dsfrButtonHarness.getButtonText());
```
Comment (flowg) [inline libs/ngx-dsfr/button/src/lib/testing/button.component.spec.ts:82]: Extra parentheses not necessary

File: libs/ngx-dsfr/button/src/lib/testing/button.component.spec.ts [hunk 7]
```diff
@@ -61,40 +59,42 @@ describe('DsfrButtonComponent', () => {
...
+
+      expect(title).toContain(label);
```
Comment (flowg) [inline libs/ngx-dsfr/button/src/lib/testing/button.component.spec.ts:84]: More like `.toBe()` ?

**Review by flowg — CHANGES_REQUESTED** [review]

File: .eslintrc.json (hunk 2 above)
[the remark this answers is not in the document]
Comment (flowg) [reply .eslintrc.json:15]: Ok, we'll see about it if and when it poses a problem 😉

**Review by flowg — APPROVED** [review]

**Pull request merged**